src/Sitewards/DBCompare/Worker/Item/AbstractItemWorker.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use Doctrine\DBAL\Connection;
+use Symfony\Component\Filesystem\Filesystem;
+use Sitewards\DBCompare\Exception\NoDiffFileNameGivenException;
+
+abstract class AbstractItemWorker
+{
+    /** @var Filesystem */
+    private $oFileSystem;
+    /** @var Connection */
+    private $oConnection;
+    /** @var string */
+    private $sDiffFileName;
+
+    /**
+     * @param Connection $oConnection
+     * @param string $sDiffFileName
+     */
+    public function __construct(Connection $oConnection, $sDiffFileName)
+    {
+        if (is_null($sDiffFileName)) {
```

Comment:
should I check for more that just null here? I could check that it is a string and that it is a valid file name?